app/Database/CourseInsertion.hs

```diff
@@ -11,16 +11,35 @@ into the database. These functions are used as helpers for the WebParsing module
 module Database.CourseInsertion
     (insertCourse,
      setTutorialEnrolment,
-     setPracticalEnrolment) where
+     setPracticalEnrolment,
+     saveGraphJSON) where
 
 import qualified Data.Text as T
-
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import Happstack.Server.SimpleHTTP
 import Control.Monad.IO.Class (MonadIO)
 import Control.Monad.Trans.Reader (ReaderT)
 import Data.Maybe (fromMaybe)
-
-import Database.Persist.Sqlite (insert_, SqlBackend, (=.), (==.), updateWhere)
+import Config (databasePath)
+import Database.Persist.Sqlite (selectFirst, fromSqlKey, toSqlKey, insertMany_, insert_, insert, SqlBackend, (=.), (==.), updateWhere, runSqlite)
 import Database.Tables
+import Data.Aeson
+
+-- | Inserts SVG graph data into Texts, Shapes, and Paths tables
+saveGraphJSON :: String -> String -> IO Response
+saveGraphJSON jsonStr nameStr = do
+    let jsonObj = decode $ BSL.pack jsonStr
+    case jsonObj of
+        Nothing -> return $ toResponse ("Error" :: String)
+        Just (SvgJSON texts shapes paths) -> do
+            response <- runSqlite databasePath $ do insertMany_ texts
+                                                    insertMany_ shapes
+                                                    insertMany_ paths
+                                                    gId <- insert $ Graph nameStr 256 256
```

Comment:
You're on the right track, but you should really use record updating.

If you have a text `t` and you want to change the graph field, keep in mind that the actual name of the field is `textGraph`, so you can do something like `t { textGraph = gId }`. Do that for all of `texts`, `shapes`, and `paths` before inserting into the database, so you won't have to later do an update.